pkg/gatherers/conditional/gather_api_request_count.go

```diff
@@ -0,0 +1,107 @@
+package conditional
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/insights-operator/pkg/gatherers"
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/openshift/insights-operator/pkg/utils"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+// APIRequestCount defines a type used when marshaling into JSON
+type APIRequestCount struct {
+	ResourceName        string `json:"resource"`
+	RemovedInRelease    string `json:"removed_in_release"`
+	TotalRequestCount   int64  `json:"total_request_count"`
+	LastDayRequestCount int64  `json:"last_day_request_count"`
+}
+
+// BuildGatherApiRequestCounts creates a gathering closure which collects API requests counts for the
+// resources mentioned in the alert provided as a string parameter
+// Params is of type AlertIsFiringConditionParams:
+//   - alert_name string - name of the firing alert
+//
+// * Location in archive: conditional/alerts/<alert_name>/api_request_counts.json
+// * Since versions:
+//   * 4.10+
+func (g *Gatherer) BuildGatherAPIRequestCounts(paramsInterface interface{}) (gatherers.GatheringClosure, error) {
+	params, ok := paramsInterface.(GatherAPIRequestCountsParams)
+	if !ok {
+		return gatherers.GatheringClosure{}, fmt.Errorf(
+			"unexpected type in paramsInterface, expected %T, got %T",
+			GatherAPIRequestCountsParams{}, paramsInterface,
+		)
+	}
+
+	return gatherers.GatheringClosure{
+		Run: func(ctx context.Context) ([]record.Record, []error) {
+			dynamicClient, err := dynamic.NewForConfig(g.gatherProtoKubeConfig)
+			if err != nil {
+				return nil, []error{err}
+			}
+			records, errs := g.gatherAPIRequestCounts(ctx, dynamicClient, params.AlertName)
+			if errs != nil {
+				return records, errs
+			}
+			return records, nil
+		},
+		CanFail: canConditionalGathererFail,
+	}, nil
+}
+
+func (g *Gatherer) gatherAPIRequestCounts(ctx context.Context,
+	dynamicClient dynamic.Interface, alertName string) ([]record.Record, []error) {
+	resources := make(map[string]struct{})
+	for _, labels := range g.firingAlerts[alertName] {
+		resourceName := fmt.Sprintf("%s.%s.%s", labels["resource"], labels["version"], labels["group"])
+		resources[resourceName] = struct{}{}
+	}
+
+	gvr := schema.GroupVersionResource{Group: "apiserver.openshift.io", Version: "v1", Resource: "apirequestcounts"}
```

Comment:
I'm always confused about this one... 'cause we do have some resources declared inside the `const.go`, but in some cases we have it declared on its own gather file. Which is the right place?

Reply:
Yes. Good point. Well we don't have any `const.go` in the conditional gatherer yet and we don't want to introduce a dependency to `clusterconfig` one..... I guess

Reply:
Yeh, I don't block the merge, I already approved it, but... it would be nice if we could decide where do we want to store these declarations. Maybe we could change it in some other PR later.